mitiq/lre/inference/multivariate_richardson.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Functions for multivariate richardson extrapolation as defined in
+:cite:`Russo_2024_LRE`.
+"""
+
+import warnings
+from itertools import product
+from typing import Any, List, Optional, Tuple
```

Comment:
it might be better to use the built-in default types here

Reply:
I agree with this comment in general, but most of `mitiq` uses this pattern. I think it might be good to have a separate issue that just covers using native types for typing throughout the whole codebase and keeping this as is (for now) for consistency.

Reply:
@vprusso You missed an earlier discussion on this: https://github.com/unitaryfund/mitiq/pull/2464#discussion_r1718343500